types/src/transaction/pricing_mode.rs

```diff
@@ -0,0 +1,122 @@
+use alloc::vec::Vec;
+use core::fmt::{self, Display, Formatter};
+
+#[cfg(feature = "datasize")]
+use datasize::DataSize;
+#[cfg(any(feature = "testing", test))]
+use rand::Rng;
+#[cfg(feature = "json-schema")]
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+#[cfg(doc)]
+use super::Transaction;
+use crate::bytesrepr::{self, FromBytes, ToBytes, U8_SERIALIZED_LENGTH};
+#[cfg(any(feature = "testing", test))]
+use crate::testing::TestRng;
+
+const GAS_PRICE_MULTIPLIER_TAG: u8 = 0;
+const NOOP_TAG: u8 = 1;
+const RESERVED_TAG: u8 = 2;
+
+/// The pricing mode of a [`Transaction`].
+#[derive(Clone, Ord, PartialOrd, Eq, PartialEq, Hash, Serialize, Deserialize, Debug)]
+#[cfg_attr(feature = "datasize", derive(DataSize))]
+#[cfg_attr(
+    feature = "json-schema",
+    derive(JsonSchema),
+    schemars(description = "Pricing mode of a Transaction.")
+)]
+#[serde(deny_unknown_fields)]
+#[non_exhaustive]
+pub enum PricingMode {
+    /// Multiplies the gas used by the given amount.
+    ///
+    /// This is the same behaviour as for the `Deploy::gas_price`.
+    GasPriceMultiplier(u64),
+    /// Irrelevant.
+    Noop,
```

Comment:
Should this be perhaps "Fixed" for FIFO lanes?

Reply:
Happy to change it just now.  I think we have plenty of wiggle room to modify it again in the near future if required.

Reply:
Done in [633530d](https://github.com/casper-network/casper-node/pull/4130/commits/633530d1a7ad8d8957b9363ae6935a9318d6694c).